src/main/java/com/technicjelle/bluemapofflineplayermarkers/MarkerHandler.java

```diff
@@ -155,17 +158,24 @@ public void loadOfflineMarkers() {
 
 			//Collect data
 			int gameModeInt = (int) nbtData.get("playerGameType").getValue();
-			long worldUUIDLeast = (long) nbtData.get("WorldUUIDLeast").getValue();
-			long worldUUIDMost = (long) nbtData.get("WorldUUIDMost").getValue();
 			@SuppressWarnings("unchecked") //Apparently this is just how it should be https://discord.com/channels/665868367416131594/771451216499965953/917450319259115550
 			List<Double> position = ((List<DoubleTag>) nbtData.get("Pos").getValue()).stream().map(DoubleTag::getValue).collect(Collectors.toList());
+			World world;
+			try {
```

Comment:
This can be simplified to an optional check instead, logic using error handling isn't great.
```
Optional<Tag<?>> worldUUIDLeastOptional = Optional.ofNullable(nbtData.get("WorldUUIDLeast"));
Optional<Tag<?>> worldUUIDMostOptional = Optional.ofNullable(nbtData.get("WorldUUIDMost"));

if (worldUUIDLeastOptional.isEmpty() || worldUUIDMostOptional.isEmpty()) {
// old flow
} else {
// normal flow
long worldUUIDLeast = (long) worldUUIDLeastOptional.get().getValue();
long worldUUIDMost = (long) worldUUIDMostOptional.get().getValue();
}

```

Reply:
I can switch to this approach, but it will be several days before I have time to apply and test it.

Reply:
I had fixed this on my end before I saw your PR, hence why I mentioned it. I can fork your repo and PR into this as well. Still need to await @TechnicJelle 's response anyway.

Reply:
@z-glitch https://github.com/z-glitch/BlueMapOfflinePlayerMarkers/pull/1